### PR TITLE
13604: Yet another attempt to get a top level frame origin for shields.

### PR DESCRIPTION
**Pull request by iefremov**
The current way doesn't work, because it is basically not a
top level origin, but "Site", aka eTLD+1. So we cannot use for
checking shield permissions.

Fix https://github.com/brave/brave-browser/issues/13533
Fix https://github.com/brave/brave-browser/issues/13604

## Submitter Checklist:

- [x] There is a [ticket](https://github.com/brave/brave-browser/issues) for my issue.
- [x] Used Github [auto-closing keywords](https://help.github.com/articles/closing-issues-via-commit-messages/) in the commit message.
- [x] Wrote a good [PR/commit description](https://google.github.io/eng-practices/review/developer/cl-descriptions.html)
- [ ] Added appropriate labels (`QA/Yes` or `QA/No`; `release-notes/include` or `release-notes/exclude`; `OS/...`) to the associated issue
- [x] Checked the PR locally: `npm run test -- brave_browser_tests`, `npm run test -- brave_unit_tests`, `npm run lint`, `npm run gn_check`, `npm run tslint`
- [x] Ran `git rebase master` (if needed).
- [ ] Requested a security/privacy review as needed.

## Reviewer Checklist:

- [ ] New files have MPL-2.0 license header.
- [ ] Adequate test coverage exists to prevent regressions
- [ ] Major classes, functions and non-trivial code blocks are well-commented
- [ ] Changes in component dependencies are properly reflected in `gn`
- [ ] Code follows the [style guide](https://chromium.googlesource.com/chromium/src/+/HEAD/styleguide/c++/c++.md)
- [ ] Test plan is specified in PR before merging

## After-merge Checklist:

- [ ] The associated issue milestone is set to the smallest version that the
  changes has landed on.
- [ ] All relevant documentation has been updated, for instance:
  - [ ] https://github.com/brave/brave-browser/wiki/Deviations-from-Chromium-(features-we-disable-or-remove)
  - [ ] https://github.com/brave/brave-browser/wiki/Proxy-redirected-URLs
  - [ ] https://github.com/brave/brave-browser/wiki/Fingerprinting-Protections
  - [ ] https://github.com/brave/brave-browser/wiki/Brave%E2%80%99s-Use-of-Referral-Codes
  - [ ] https://github.com/brave/brave-browser/wiki/Custom-Headers
  - [ ] https://github.com/brave/brave-browser/wiki/Web-Compatibility-Exceptions-in-Brave
  - [ ] https://github.com/brave/brave-browser/wiki/QA-Guide
  - [ ] https://github.com/brave/brave-browser/wiki/P3A

### CODE REVIEW

File: browser/net/url_context.cc [hunk 1]
```diff
@@ -98,7 +98,7 @@ std::shared_ptr<brave::BraveRequestInfo> BraveRequestInfo::MakeCTX(
     // given that this is always empty for subresources.
     ctx->network_isolation_key =
         request.trusted_params->isolation_info.network_isolation_key();
-    ctx->tab_origin = ctx->network_isolation_key.GetTopFrameSite()
+    ctx->tab_origin = request.trusted_params->isolation_info.top_frame_origin()
```
Comment (mkarolin) [inline browser/net/url_context.cc:101]: `trusted_params` is `base::Optional` - should maybe check if it's set?

Comment (iefremov) [reply browser/net/url_context.cc:101]: we are already inside this check :)

Comment (mkarolin) [reply browser/net/url_context.cc:101]: Ah, and so we are 👍

**Review by mkarolin — APPROVED** [review]
LGTM

**Review by fmarier — APPROVED** [review]

**Comment by mkarolin** [comment]
Rebased on the latest `master`.

**Comment by mkarolin** [comment]
**CI:**
* MacOS only failed during the `dist` step due to timeout. Good to merge since the `build` step completed successfully.

**Comment by kjozwiak** [comment]
Verification **PASSED** on `Win 10 x64` using the following build:

```
Brave | 1.21.7 Chromium: 88.0.4324.87 (Official Build) nightly (64-bit)
-- | --
Revision | dd01ff8f58c65af81127ad5c105c79d5b571d8f3-refs/branch-heads/4324@{#1702}
OS | Windows 10 OS Version 2009 (Build 19042.746)
```

**`Case 1`** - https://github.com/brave/brave-browser/issues/13604

* ensured that the shield icon is disabled/turns grey and doesn't display the blocking count when disabled
* ensured that the shield icon is correctly displaying the number of blocked ads when enabled
* ensured that ads weren't being displayed under `www.google.ca` when `Trackers & ads blocked (aggressive)` was enabled
* ensured that ads are being blocked on various websites when `Trackers & ads blocked (standard)` has been selected (Default)
* ensured that selecting `Allow all trackers & ads` disabled ad blocking and ads start appearing on various websites
* ensured that `Scripts blocked` was working correctly under https://www.blizzard.com/en-us

@rebron also verified https://github.com/brave/brave-browser/issues/13604 via https://github.com/brave/brave-browser/issues/13604#issuecomment-761719059.

**`Shields Enabled`** | **`Shields Disabled`**
-----------------------|--------------------------
defaultShields | shieldsDisabled

**`Shields Enabled (Blocking Ads)`** | **`Shields Disabled (Not Blocking Ads`**
-----------------------|--------------------------
defaultShieldsAds | shieldsDisabledAds

**Pull request merged**